api/infrastructure/parallelcluster-api.yaml

```diff
@@ -0,0 +1,246 @@
+    Properties:
+      Tags:
+        parallelcluster: !Ref ParallelClusterVersion
+      StageName: prod
+      DefinitionBody:
+        Fn::Transform:
+          Name: AWS::Include
+          Parameters:
+            Location: !Ref ApiDefinitionS3Uri
+
+  APIGatewayExecutionRole:
+   Type: AWS::IAM::Role
+   Properties:
+     AssumeRolePolicyDocument:
+       Version: 2012-10-17
+       Statement:
+         - Effect: Allow
+           Principal:
+             Service:
+               - apigateway.amazonaws.com
+           Action:
+             - 'sts:AssumeRole'
+     Policies:
+       - PolicyName: lambda-invoke
+         PolicyDocument:
+           Version: '2012-10-17'
+           Statement:
+             - Effect: Allow
+               Action: lambda:InvokeFunction
+               Resource: !GetAtt ParallelClusterFunction.Arn
+
+  ParallelClusterFunction:
+    Type: AWS::Serverless::Function
+    DependsOn: LambdaLogGroup
+    Properties:
+      PackageType: Image
+      Tracing: Active
+      Tags:
+        parallelcluster: !Ref ParallelClusterVersion
+      Policies:
+        - AWSLambdaBasicExecutionRole
+        - AWSXRayDaemonWriteAccess
+        - AdministratorAccess
+      ImageUri: !If [UsesCustomEcrImageUri, !Ref EcrImageUri, !Ref PublicEcrImageUri]
+
+  LambdaLogGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      RetentionInDays: 30
+
+  InstanceRole:
+    Condition: DoesNotUseCustomEcrImageUri
+    Type: AWS::IAM::Role
+    Properties:
+      ManagedPolicyArns:
+        - Fn::Sub: arn:${AWS::Partition}:iam::aws:policy/AmazonSSMManagedInstanceCore
+        - Fn::Sub: arn:${AWS::Partition}:iam::aws:policy/EC2InstanceProfileForImageBuilderECRContainerBuilds
+      AssumeRolePolicyDocument:
+        Statement:
+          - Action:
+              - sts:AssumeRole
+            Effect: Allow
+            Principal:
+              Service:
+                - ec2.amazonaws.com
+        Version: '2012-10-17'
+      Path: /executionServiceEC2Role/
+
+  InstanceProfile:
+    Condition: DoesNotUseCustomEcrImageUri
+    Type: AWS::IAM::InstanceProfile
+    Properties:
+      Path: /executionServiceEC2Role/
+      Roles:
+        - !Ref InstanceRole
+
+  InfrastructureConfiguration:
+    Condition: DoesNotUseCustomEcrImageUri
+    Type: AWS::ImageBuilder::InfrastructureConfiguration
+    Properties:
+      Name: ParallelClusterImageBuilderInfrastructureConfiguration
+      InstanceProfileName: !Ref InstanceProfile
+      TerminateInstanceOnFailure: true
+
+  PrivateEcrRepository:
+    Condition: DoesNotUseCustomEcrImageUri
+    Type: AWS::ECR::Repository
+    Properties:
+      RepositoryName: !Sub
+        - 'aws-parallelcluster-${Version}'
+        - { Version: !Ref ParallelClusterVersion }
+
+  EcrImageRecipe:
+    Condition: DoesNotUseCustomEcrImageUri
+    Type: AWS::ImageBuilder::ContainerRecipe
+    Properties:
+      Components:
+        - ComponentArn: !Sub arn:aws:imagebuilder:${AWS::Region}:aws:component/update-linux/1.0.0
```

Comment:
what do we need this for? is it to have at least 1 component?I'm wondering what is this component updating and if we can have a no-op component instead

Reply:
Afaik yes, we need to specify at least one component and there is no no-op component. Also, in practice this should not really be used since in the template section I don't specify `{{{ imagebuilder:components }}}`. Ideally I would like to remove this section, I have reached out to the ImageBuilder team to ask for clarification

Reply:
I have validated with the ImageBuilder team that the behavior corresponds to what I have detailed above